contracts/build.rs

```diff
@@ -16,67 +15,73 @@ fn main() {
-    generate_contract(
-        "GPv2Settlement",
-        hashmap! {
-            1 => (Address::from_str("0x3328f5f2cEcAF00a2443082B657CedEAf70bfAEf").unwrap(), Some("0x34b7f9a340e663df934fcc662b3ec5fcd7cd0c93d3c46f8ce612e94fff803909".parse().unwrap())),
-            4 => (Address::from_str("0x3328f5f2cEcAF00a2443082B657CedEAf70bfAEf").unwrap(), Some("0x52badda922fd91052e6682d125daa59dea3ce5c57add5a9d362bec2d6ccfd2b1".parse().unwrap())),
-            100 => (Address::from_str("0x3328f5f2cEcAF00a2443082B657CedEAf70bfAEf").unwrap(), Some("0x95bbefbca7162435eeb71bac6960aae4d7112abce87a51ad3952d7b7af0279e3".parse().unwrap())),
-        },
-    );
-    generate_contract("GPv2AllowListAuthentication", hashmap! {});
-    generate_contract(
-        "WETH9",
-        hashmap! {
-            // Rinkeby & Mainnet Addresses are part of the artefact
-            100 => (Address::from_str("e91D153E0b41518A2Ce8Dd3D7944Fa863463a97d").unwrap(), None),
-        },
-    );
+    generate_contract("IUniswapLikeRouter");
+    generate_contract("IUniswapLikePair");
+    generate_contract_with_config("SushiswapV2Router02", |builder| {
+        builder
+            .add_deployment_str(1, "0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F")
+            .add_deployment_str(4, "0x1b02dA8Cb0d097eB8D57A175b88c7D8b47997506")
+            .add_deployment_str(100, "0x1b02dA8Cb0d097eB8D57A175b88c7D8b47997506")
+    });
+    generate_contract_with_config("SushiswapV2Factory", |builder| {
+        builder
+            .add_deployment_str(1, "0xC0AEe478e3658e2610c5F7A4A2E1777cE9e4f2Ac")
+            .add_deployment_str(4, "0xc35DADB65012eC5796536bD9864eD8773aBc74C4")
+            .add_deployment_str(100, "0xc35DADB65012eC5796536bD9864eD8773aBc74C4")
+    });
+    generate_contract("SushiswapV2Pair");
+    generate_contract_with_config("GPv2Settlement", |builder| {
+        builder
+            .with_contract_mod_override(Some("gpv2_settlement"))
+            .add_deployment(
+                1,
+                addr("0x3328f5f2cEcAF00a2443082B657CedEAf70bfAEf"),
+                Some(tx(
+                    "0x34b7f9a340e663df934fcc662b3ec5fcd7cd0c93d3c46f8ce612e94fff803909",
+                )),
+            )
+            .add_deployment(
+                4,
+                addr("0x3328f5f2cEcAF00a2443082B657CedEAf70bfAEf"),
+                Some(tx(
+                    "0x52badda922fd91052e6682d125daa59dea3ce5c57add5a9d362bec2d6ccfd2b1",
+                )),
+            )
+            .add_deployment(
+                100,
+                addr("0x3328f5f2cEcAF00a2443082B657CedEAf70bfAEf"),
+                Some(tx(
+                    "0x95bbefbca7162435eeb71bac6960aae4d7112abce87a51ad3952d7b7af0279e3",
+                )),
+            )
+    });
+    generate_contract("GPv2AllowListAuthentication");
+    generate_contract_with_config("WETH9", |builder| {
+        builder.add_deployment_str(100, "0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d")
```

Comment:
Here is what caused the issues on xDAI. There was a typo and this line was incorrectly:
```suggestion
        builder.add_deployment_str(1, "0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d")
```

Reply:
Wouldn't it have been a simpler hotfix to just make a PR that only add this 100 back in?